### PR TITLE
Consistent environment variable names for DOTNET_RUNNING_IN_CONTAINER

### PR DESCRIPTION
**Pull request by fbeltrao**
This PR is a potential fix for issue #677

- 2.1 images for windows declare both environment variables (DOTNET_RUNNING_IN_CONTAINERS and DOTNET_RUNNING_IN_CONTAINER)
- 2.2 and 3.0 images declare only DOTNET_RUNNING_IN_CONTAINER (breaking change)

### CODE REVIEW

File: 2.1/aspnetcore-runtime/nanoserver-1709/amd64/Dockerfile [hunk 1]
```diff
@@ -33,4 +33,6 @@ USER ContainerUser
 # Configure web servers to bind to port 80 when present
 ENV ASPNETCORE_URLS=http://+:80 `
     # Enable detection of running in a container
-    DOTNET_RUNNING_IN_CONTAINERS=true
+    DOTNET_RUNNING_IN_CONTAINERS=true `
```
Comment (MichaelSimons) [inline 2.1/aspnetcore-runtime/nanoserver-1709/amd64/Dockerfile]: I'm of the opinion that we should list the one we want people to use first.  Then on the incorrect one, I think a comment similar to the following should be added.

```
# Deprecated, use `DOTNET_RUNNING_IN_CONTAINER` instead - https://github.com/dotnet/dotnet-docker/issues/677
```

Comment (fbeltrao) [reply 2.1/aspnetcore-runtime/nanoserver-1709/amd64/Dockerfile]: Makes sense. I will make the changes

Comment (fbeltrao) [reply 2.1/aspnetcore-runtime/nanoserver-1709/amd64/Dockerfile]: Changes done, testing has started

**Comment by MichaelSimons** [comment]
@fbeltrao - Thanks for the contribution.

**Pull request merged**